test/extended/cli/builds.go

```diff
@@ -0,0 +1,179 @@
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	"k8s.io/kubectl/pkg/util/slice"
+)
+
+const (
+	bcOutputTemplate      = "{{with .spec.output.to}}{{.kind}} {{.name}}{{end}}"
+	testDockerFileContent = "FROM image-registry.openshift-image-registry.svc:5000/openshift/tests:latest"
+)
+
+type expectedBuild struct {
+	bcName      string
+	isName      string
+	sourceType  string
+	outputType  string
+	outputImage string
+}
+
+func (r *expectedBuild) setDefaults() {
+	if r.outputType == "ImageStreamTag" && r.isName == "" {
+		r.isName = r.bcName
+	}
+}
+
+func testNewBuild(oc *exutil.CLI, opts expectedBuild) {
+	opts.setDefaults()
+	// new build
+	buildOut, err := oc.Output()
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(buildOut).To(o.And(
+		o.ContainSubstring(fmt.Sprintf("buildconfig.build.openshift.io \"%v\" created", opts.bcName)),
+		o.ContainSubstring("Success"),
+	))
+	if opts.sourceType == "Dockerfile" {
+		o.Expect(buildOut).To(o.ContainSubstring(fmt.Sprintf("imagestream.image.openshift.io \"tests\" created"))) // see testDockerFileContent
+	}
+	if opts.isName != "" {
+		o.Expect(buildOut).To(o.ContainSubstring(fmt.Sprintf("imagestream.image.openshift.io \"%v\" created", opts.isName)))
+	}
+	// check source type
+	out, err := oc.Run("get").Args("bc/" + opts.bcName).Output()
```

Comment:
What would happen if opts.bcName is empty? Would we error out or ignore?

I'd rather be explicit in the individual test rather than trying to get arg and then do validation here.

Reply:
we could add a check there

Reply:
I am using different functions now and I added checks for bcName empty, but it seems superfluous now in these...